climbing_stairs/climbing_stairs.py

```diff
@@ -1,14 +1,17 @@
 #!/usr/bin/python
 
 import sys
-
+# can go up one stair at a time
+# two stairs at a time
+# three stairs at a time
+# and other random orders
 def climbing_stairs(n, cache=None):
-  pass 
+    options = [1, 2, 3]
```

Comment:
nice idea here with caching you're definitely on the right track and your solution will be optimized from the get go.